distribution/src/config/jvm.options

```diff
@@ -36,8 +36,8 @@
 # Xms represents the initial size of the JVM heap
 # Xmx represents the maximum size of the JVM heap
 
--Xms${heap.min}
--Xmx${heap.max}
+# -Xms${heap.min}
```

Comment:
Still need to think about how to change this. Should we remove this section entirely? Continue to include it also recommend folks stick with auto-determined heap size?

Reply:
We may want to leave it in with a note, since until now our users have always had to configure it, and our training courses will expect it to exist. Maybe we can remove it in master but retain it with a comment in `7.x`?

Reply:
It's tricky. Either way (removing, or commenting out), we are going to potentially break automation here. For example, if there's automation in place that downloads our tar archive, explodes it, and uses `sed` to replace the heap size.

That leads me to wonder out loud if we should not break users here. I'd propose the following: in 7.x, we leave this as-is in the jvm.options file. Perhaps we provide a warning when the heap size is set manually by a user, pointing them to documentation for machine-dependent heap sizes. In Cloud, we still want to activate the machine-dependent heap sizes, so we'd need deployments there to *remove* the `-Xms/-Xmx` flags from their jvm.options file.

In 8.0, we remove this from the default jvm.options file.

Reply:
I understand the concern, though I'm not sure about a warning. I don't think we intend to *disallow* setting the heap manually right? Yet a warning for setting the heap would seem like just that. Such a sed replacement in practice would seem fragile. It could break on any change to the default. 

I'm conflicted. I don't see a way to warn some users without annoying others. But I do think this is something we can followup on, so in this PR I will restore the original value.

Reply:
My suggestion was:

>Perhaps we provide a warning when the heap size is set manually by a user, pointing them to documentation for machine-dependent heap sizes.

Not a warning that we will disallow setting the heap size manually, but a permanent warning that the heap size is being set manually and we recommend not for the optimal experience.

>But I do think this is something we can followup on, so in this PR I will restore the original value.

Are you saying then that the default experience will be to not get machine-dependent heap sizing? That seems contrary to our goals here.

Reply:
If we leave the jvm.options as is, the default will remain not getting machine-dependent heap sizing. If we emit a warning, any user who chooses to not want to use machine-dependent heap sizing will get an annoying warning they cannot use. Do we agree on those statements?

My suggestion was to leave the jvm.options as is (so as not to break the theoretical users that could be using sed to replace the heap size), and then follow up with whatever we decide to do regarding helping users convert within 7.x. This suggestion is because cloud will not rely on the default.

Reply:
> If we leave the jvm.options as is, the default will remain not getting machine-dependent heap sizing. If we emit a warning, any user who chooses to not want to use machine-dependent heap sizing will get an annoying warning they cannot use. Do we agree on those statements?

Why can't they use it? The point of the warning is to tell them to stop setting the heap size manually, let Elasticsearch do it for them. The warning is useful to them.

Reply:
Sorry, not sure how "cannot use" go in my response, it was meant to be "cannot remove". That is it is a persistent warning that the only way to remove is to go with ES's memory choices. The warning is useful in that they find out they *could* be using our automatic heap sizing, but are we saying users will never tune heap anymore? I personally get annoyed if there are warnings I cannot remove with normal usage.

Reply:
Jason and I spoke and came to an agreement on the above discussion. @jasontedor Let me know if I missed or misunderstood anything.

We should add a warning when heap size is manually specified, since our goal is to have users use the auto heap. There are however some issues that prevent us from doing that in this PR. First is a technical issue: how do we pass back the message when the only output of the jvm options parser is the final jvm options?. The second issue is allowing cloud time to adapt to using auto sizing so that we do not spew errors across all cloud deployments. I created https://github.com/elastic/elasticsearch/issues/66308 to ensure we followup with adding this warning.

Regarding whether to leave the heap options or remove them in a minor, we think the auto heap sizing is important enough that we don't want to have any more new users specifying heap. So, we should remove the Xmx/Xms from `jvm.options`. Users should no longer be relying on modifying `jvm.options`, but instead specifying these in `jvm.options.d` files.

Reply:
@mark-vieira @jasontedor I've removed Xmx/Xms in 88a81a11a80cc2805b74913c0f337bc28c4c4474, but left in a comment explaining where the heap size should be set if it needs to be.